src/main/java/com/rultor/agents/github/CommentsTag.java

```diff
@@ -94,44 +106,191 @@ public CommentsTag(final Github ghub) {
+        rel.name(issue.title());
+        rel.prerelease(true);
+        rel.body(
+            String.format(
+                // @checkstyle LineLength (1 line)
+                "See #%d, release log:\n\n%s\n\nReleased by Rultor %s, see [build log](%s)",
+                issue.number(),
+                new CommitsLog(repo).build(prev, rel.publishedAt()),
+                Manifests.read("Rultor-Version"), home
+            )
+        );
+        Logger.info(this, "tag %s created and commented", tag);
+    }
+
+    /**
+     * Logs an issue because the new tag is too old.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag that for the new release.
+     * @param previous The previous releases.
+     * @throws IOException If there is a problem.
+     */
+    private void logTagTooOld(final Issue issue, final String tag,
+        final List<DefaultArtifactVersion> previous) throws IOException {
+        issue.comments().post(
+            String.format(
+                // @checkstyle LineLength (1 line)
+                CommentsTag.PHRASES.getString("CommentsTag.version-too-low"),
+                tag, previous.toString()
+            )
+        );
+        Logger.info(
+            this, "tag %s must be greater than previous version %s",
+            tag, previous
+        );
+    }
+
+    /**
+     * Logs a comment because the tag is invalid.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag which is invalid.
+     * @throws IOException If there is a problem.
+     */
+    private void logInvalidTag(final Issue issue, final String tag)
+        throws IOException {
+        issue.comments().post(
+            String.format(
+                CommentsTag.PHRASES.getString("CommentsTag.invalid-tag"), tag
+            )
+        );
+        Logger.info(this, "tag %s is invalid", tag);
+    }
+
+    /**
+     * Logs a comment because the tag has already been released.
+     * @param issue The Issue to add the comment to.
+     * @param tag The tag that has already been released.
+     * @param rels The other releases.
+     * @param home The URI of home.
+     * @throws IOException If there is a problem.
+     * @checkstyle ParameterNumberCheck (15 lines)
+     */
+    private void logDuplicateTag(final Issue.Smart issue, final String tag,
+        final Releases.Smart rels, final URI home) throws IOException {
+        final Release.Smart rel = new Release.Smart(rels.find(tag));
+        rel.body(
+            String.format(
+                "%s\n\nSee also #%d and [build log](%s)",
+                rel.body(), issue.number(), home
+            )
+        );
+        issue.comments().post(
+            String.format(
+                CommentsTag.PHRASES.getString("CommentsTag.duplicate"),
+                tag
+            )
+        );
+        Logger.info(this, "duplicate tag %s commented", tag);
+    }
+
+    /**
+     * Valid version numbers:
+     * .1
```

Comment:
@gumbelmj I don't think this is a valid version. 

Also the method JavaDoc should state that this method validates version numbers instead of giving the reader a bunch of examples to figure it out from before reading on to the return tag.